packages/SwingSet/misc-tools/store-stats/src/store-stats.js

```diff
@@ -0,0 +1,160 @@
+    const prepare = (strings, ...params) => {
+      const dml = strings.join('?');
+      return { stmt: db.prepare(dml), params };
+    };
+    const sql = (strings, ...args) => {
+      const { stmt, params } = prepare(strings, ...args);
+      return stmt.all(...params);
+    };
+    sql.get = (strings, ...args) => {
+      const { stmt, params } = prepare(strings, ...args);
+      return stmt.get(...params);
+    };
+    sql.iterate = (strings, ...args) => {
+      const { stmt, params } = prepare(strings, ...args);
+      return stmt.iterate(...params);
+    };
+    return sql;
+  };
+
+  const sql = dbTool();
+
+  /** @param {string} key */
+  const kvGet = key => sql.get`select * from kvStore where key = ${key}`.value;
+
+  const kvAll = key => sql.iterate`select * from kvStore where key GLOB ${key}`;
+
+  return Object.freeze({
+    findHeight: () => {
+      console.log('Fetching block height from the database');
+      return kvGet('host.height');
+    },
+    findVatItems: () => {
+      console.log('Fetching vat items from the database');
+      return kvAll('v[0-9]*.*');
+    },
+  });
+};
+
+export const makeStatsSender = () => {
+  const sendMetrics = data => {
+    const { height, vatStats } = data;
+    console.log(
+      `Sending metrics of ${vatStats.size} vats to datadog for block ${height}`,
+    );
+    for (const [vatID, stats] of vatStats) {
+      const tags = [`height:${height}`, `vatID:${vatID}`];
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_exports_reachable',
+        stats.objectExportsReachable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_exports_recognizable',
+        stats.objectExportsRecognizable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_imports_reachable',
+        stats.objectImportsReachable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.object_imports_recognizable',
+        stats.objectImportsRecognizable,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.promises',
+        stats.promises,
+        tags,
+      );
+      dogstatsd.gauge(
+        'swingset.kvstore.vat_stats.vat_store_keys',
+        stats.vatStoreKeys,
+        tags,
+      );
+      dogstatsd.increment('swingset.kvstore.stats');
+    }
+    dogstatsd.close();
+    console.log('Done sending metrics to datadog');
+  };
+
+  return Object.freeze({
+    sendMetrics,
+  });
+};
+
+export const makeStatsTransformer = () => {
+  // kvstore has pairs like:
+  // "v$NN.c.ko45": "R o+4"      # reachable export
+  // "v$NN.c.ko46": "_ o+5"      # merely-recognizable export
+  // "v$NN.c.ko47": "R o-6"      # reachable import
+  // "v$NN.c.ko48": "_ o-7"      # merely-recognizable import
+  // "v$NN.c.kp70": "R p+12"      # promise (always "R", sometimes "p+" sometimes "p-")
+
+  const object = /^v\d+\.c\..*ko\d.*$/; // vNN.c.koNNNNN*
```

Comment:
I think the trailing `.*$` is redundant, might consider omitting it (although I imagine the regexp compiler discards it, so it's not like it's likely to make anything slower, and it does make it nicely clear what we're expecting).